From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: [PATCH] abort needs cstdlib

Fix this compile error when compiled --without-exceptions:

libtool: compile:  arm-1136jfs-linux-gnueabi-g++ -DHAVE_CONFIG_H -I. -I.. -I../inc -isystem /home/rsc/svn/oselas/bsp/pengutronix/OSELAS.BSP-Pengutronix-AllYes-trunk/platform-ARM-reference/sysroot-target/include -isystem /home/rsc/svn/oselas/bsp/pengutronix/OSELAS.BSP-Pengutronix-AllYes-trunk/platform-ARM-reference/sysroot-target/usr/include -I../src -DCCXX_EXPORT_LIBRARY -D_GNU_SOURCE -fno-exceptions -fno-rtti -fno-check-new -finline -I../inc -g -O2 -MT mutex.lo -MD -MP -MF .deps/mutex.Tpo -c mutex.cpp  -fPIC -DPIC -o .libs/mutex.o
mutex.cpp: In constructor 'ost::Conditional::Conditional(const char*)':
mutex.cpp:145: error: 'abort' was not declared in this scope

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---
# rsc: https://savannah.gnu.org/patch/index.php?7198

 src/engine.cpp    |    1 +
 src/mempager.cpp  |    1 +
 src/mutex.cpp     |    1 +
 src/semaphore.cpp |    1 +
 4 files changed, 4 insertions(+)

Index: commoncpp2-1.8.0/src/mutex.cpp
===================================================================
--- commoncpp2-1.8.0.orig/src/mutex.cpp
+++ commoncpp2-1.8.0/src/mutex.cpp
@@ -44,6 +44,7 @@
 #include <cc++/slog.h>
 #include <iostream>
 #include <cerrno>
+#include <cstdlib>
 
 #ifdef HAVE_GCC_CXX_BITS_ATOMIC
 using namespace __gnu_cxx;
Index: commoncpp2-1.8.0/src/semaphore.cpp
===================================================================
--- commoncpp2-1.8.0.orig/src/semaphore.cpp
+++ commoncpp2-1.8.0/src/semaphore.cpp
@@ -42,6 +42,7 @@
 #include <cc++/thread.h>
 #include "private.h"
 #include <cstdio>
+#include <cstdlib>
 
 #ifdef	CCXX_NAMESPACES
 namespace ost {
Index: commoncpp2-1.8.0/src/mempager.cpp
===================================================================
--- commoncpp2-1.8.0.orig/src/mempager.cpp
+++ commoncpp2-1.8.0/src/mempager.cpp
@@ -41,6 +41,7 @@
 #include <cc++/slog.h>
 #include <cc++/thread.h>
 #include <cc++/misc.h>
+#include <cstdlib>
 #include "private.h"
 
 #ifdef	CCXX_NAMESPACES
Index: commoncpp2-1.8.0/src/engine.cpp
===================================================================
--- commoncpp2-1.8.0.orig/src/engine.cpp
+++ commoncpp2-1.8.0/src/engine.cpp
@@ -39,6 +39,7 @@
 #include <cc++/config.h>
 #include <cc++/string.h>
 #include <cc++/exception.h>
+#include <cstdlib>
 
 #if !defined(_MSC_VER) || _MSC_VER >= 1300
 
